db/crud.go

```diff
@@ -56,6 +56,29 @@ func (db *DatabaseContext) GetDoc(docid string) (*document, error) {
 	return doc, nil
 }
 
+func (db *DatabaseContext) GetDocSyncData(docid string) (syncData, error) {
```

Comment:
When we merge this to master, we'll need to include some 'if db.UseXattrs' handling, similar to what we're doing in GetDoc.